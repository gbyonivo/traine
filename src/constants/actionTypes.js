export const FETCH_DATA = 'FETCH_DATA';
export const DONE_FETCHING_DATA = 'DONE_FETCHING_DATA';
export const ERROR_FETCHING_DATA = 'ERROR_FETCHING_DATA';
