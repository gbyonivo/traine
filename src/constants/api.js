export const HOST = 'http://api.openweathermap.org/data/2.5';
export const API_KEY = 'a998de9d3bb97fca33d1d65014184648';