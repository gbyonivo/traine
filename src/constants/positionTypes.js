export const MOVING = 'MOVING';
export const HALTED = 'HALTED';
export const NOT_STARTED = 'NOT_STARTED';