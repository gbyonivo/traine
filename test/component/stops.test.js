import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import Stops from '../../src/components/stops';
import { stops } from '../__testData__';
import { MOVING } from '../../src/constants/positionTypes';

describe('Stops should render correctly', () => {
  jasmine.clock().install();
  jasmine.clock().mockDate(new Date(2015, 3, 2, 6, 29, 0));

  it('should render Correctly', () => {
    const actual = create(<Stops
      stops={stops}
      currentLocation={{ station: 'VXH', positionType: MOVING }}
    />);
    expect(actual).toMatchSnapshot();
  });
  
  jasmine.clock().uninstall()
});