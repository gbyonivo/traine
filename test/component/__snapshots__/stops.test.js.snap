// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stops should render correctly should render Correctly 1`] = `
<ul
  className="stops"
>
  <li
    className="
      stop
      "
  >
    <div
      className="time"
    >
      00:27
    </div>
    <div
      className="locationAndStatus"
    >
      <div
        className="location"
      >
        WAT
      </div>
      <div
        className="status"
      >
        On Time
      </div>
    </div>
  </li>
  <li
    className="
      stop
      "
  >
    <div
      className="time"
    >
      00:30
    </div>
    <div
      className="locationAndStatus"
    >
      <div
        className="location"
      >
        VXH
      </div>
      <div
        className="status"
      >
        On Time
      </div>
    </div>
  </li>
  <li
    className="
      stop
      "
  >
    <div
      className="time"
    >
      00:35
    </div>
    <div
      className="locationAndStatus"
    >
      <div
        className="location"
      >
        CLJ
      </div>
      <div
        className="status"
      >
        On Time
      </div>
    </div>
  </li>
  <li
    className="
      stop
      HALTED"
  >
    <div
      className="time"
    >
      01:13
    </div>
    <div
      className="locationAndStatus"
    >
      <div
        className="location"
      >
        WOK
      </div>
      <div
        className="status"
      >
        On Time
      </div>
    </div>
  </li>
</ul>
`;
